guava/src/com/google/common/eventbus/EventBus.java

```diff
@@ -29,6 +29,59 @@
 /**
  * Dispatches events to listeners, and provides ways for listeners to register themselves.
  *
+ * <h2>Avoid EventBus</h2>
+ *
+ * <p><b>We recommend against using EventBus.</b> It was designed many years ago, and newer
+ * libraries offer better ways to decouple components and react to events.
+ *
+ * <p>To decouple components, we recommend a dependency-injection framework. For Android code, most
+ * apps use <a href="https://dagger.dev">Dagger</a>. For server code, common options include <a
+ * href="https://github.com/google/guice/wiki/Motivation">Guice</a> and <a
+ * href="https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-introduction">Spring</a>.
+ * Frameworks typically offer a way to register multiple listeners independently and then request
+ * them together as a set (<a href="https://dagger.dev/dev-guide/multibindings">Dagger</a>, <a
+ * href="https://github.com/google/guice/wiki/Multibindings">Guice</a>, <a
+ * href="https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-autowired-annotation">Spring</a>).
+ *
+ * <p>To react to events, we recommend <a
+ * href="https://github.com/ReactiveX/RxJava/wiki">RxJava</a>, supplemented with its <a
```

Comment:
These days [Project Reactor](https://projectreactor.io/) is a popular alternative to RxJava. Worth mentioning?

Reply:
Good call. Also done.